crates/topos-clock/src/time.rs

```diff
@@ -0,0 +1,103 @@
+use std::{
+    sync::{
+        atomic::{AtomicU64, Ordering},
+        Arc,
+    },
+    time::Duration,
+};
+
+use chrono::{DateTime, Utc};
+use tokio::{
+    spawn,
+    sync::broadcast,
+    time::{interval_at, Instant},
+};
+use tracing::info;
+
+use crate::{Clock, Error, Event, BROADCAST_CHANNEL_SIZE};
+
+/// Time based clock implementation.
```

Comment:
```suggestion
/// Time based clock implementation, mostly intended for tests.
```

Right?

Reply:
Not really, until we have blockchain integration it will be the only implementation.